tracing-subscriber/src/fmt/format/json.rs

```diff
@@ -209,7 +209,10 @@ where
 
             let current_span = if self.format.display_current_span || self.format.display_span_list
             {
-                ctx.ctx.current_span().id().and_then(|id| ctx.ctx.span(id))
+                event
+                    .parent()
+                    .and_then(|id| ctx.ctx.span(id))
+                    .or_else(|| ctx.ctx.current_span().id().and_then(|id| ctx.ctx.span(id)))
```

Comment:
This is essentially re-implementing [`LookupSpan::lookup_current`](https://docs.rs/tracing-subscriber/0.2.17/tracing_subscriber/fmt/struct.FmtContext.html#method.lookup_current) --- I know this code was already there, but while we're here, why don't we just change this to
```suggestion
                    .or_else(|| ctx.lookup_current())
```